src/main/scala/recfun/RecFun.scala

```diff
@@ -15,16 +17,47 @@ object RecFun extends RecFunInterface {
    * Exercise 1
    */
   def pascal(c: Int, r: Int): Int = {
+//    if (c == 0 || c == r) 1
+//    else pascal(c-1, r-1) + pascal(c, r-1)
+    @tailrec
+    def pascalAux(currCol: Int, currRow: Int, prevTriangleRow: Array[Int], currTriangleRow: Array[Int]): Int = {
+      currTriangleRow(currCol) = (if (currCol > 0) prevTriangleRow(currCol - 1) else 0) + (if (currCol < currRow) prevTriangleRow(currCol) else 0)
+
+      if ((currCol == c) && (currRow == r)) currTriangleRow(currCol)
+      else if (currCol < currRow) pascalAux(currCol + 1, currRow, prevTriangleRow, currTriangleRow)
+      else pascalAux(0, currRow + 1, currTriangleRow, new Array(_length = currRow + 2))
+    }
```

Comment:
Over complicated solution, try to find a simpler solution.
Generally most of the tasks of coursera can be implemented in a short, smart and efficient way

Reply:
if you look at the commented code which I left on purpose, that is my simple clean solution. only problem is that it is not tail recursion :)

Reply:
if you have a shorter solution for a tail recursive implementation of this problem, I'd be happy to see!
What I did here is build (as the "accumulator" of the tail recursion) the last and current rows of the pascal triangle in order compute the current iteration's row and column